### ISSUE TITLE
[DOC]: optimize readme of ChatGPT

### ISSUE THREAD

**Issue by ver217** [hpcaitech/ColossalAI#2722]
### 📚 The doc issue

A new user is hard to start training after reading the readme.

I list some FAQs which users may concern:

- How to prepare training dataset? What does the dataset look like?
- How to save models?
- How to use the fine-tuned models to generate?
- How to use LoRA? How does it affect training?

**Comment by feifeibear** [comment]
The biggest question in my mind is:
How to initialize optimizer states when finetuning a pre-trained model?

**Comment by Gy-Lu** [comment]
Hi, I found this [picture](https://raw.githubusercontent.com/hpcaitech/public_assets/main/applications/chatgpt/experience.jpg) a little confusing. I think if there is some description, it would be much better.

**Comment by feifeibear** [comment]
> Hi, I found this [picture](https://raw.githubusercontent.com/hpcaitech/public_assets/main/applications/chatgpt/experience.jpg) a little confusing. I think if there is some description, it would be much better.

Yes, at least tell the readers: the green part is the trainable parameters and the rest parts are frozen parameters.

**Comment by Fazziekey** [comment]
I think we need a detail blog to tell user what is reward model, critic model, sft-model

**Comment by Gy-Lu** [comment]
This [picture](https://raw.githubusercontent.com/hpcaitech/public_assets/main/applications/chatgpt/chatgpt.png) is from InstructGPT, maybe there should be copyright information?

**Comment by Fazziekey** [comment]
I think instructs LM is better than chatgpt, chatgpt is a serving, instruct LM is an algorithm

**Comment by ht-zhou** [comment]
> This [picture](https://raw.githubusercontent.com/hpcaitech/public_assets/main/applications/chatgpt/chatgpt.png) is from InstructGPT, maybe there should be copyright information?

<img width="1188" alt="image" src="https://user-images.githubusercontent.com/70618399/218957253-ce9bb85e-4406-43dc-96e0-cee492390cca.png">
Thanks for your remind, and we mention the reference here.

**Comment by cryoco** [comment]
> > Hi, I found this [picture](https://raw.githubusercontent.com/hpcaitech/public_assets/main/applications/chatgpt/experience.jpg) a little confusing. I think if there is some description, it would be much better.
> 
> Yes, at least tell the readers: the green part is the trainable parameters and the rest parts are frozen parameters.

I wonder why the 4 workflows below are marked as "inference", whereas some of them are still trainable?

**Comment by binmakeswell** [comment]
##Todo
- implement PPO training and fine-tuning

We should also mentioned PPO fine-tuning. This looks a lot like we haven't implemented finetune function yet, because we plan to implement PPO-ptx fine-tuning.

**Comment by binmakeswell** [comment]
<img width="1268" alt="c56c4ea7-c449-40ba-8b53-84d6cb9e647c" src="https://user-images.githubusercontent.com/61670638/218991858-16733817-9236-4534-b994-7f63aed2dcf7.png">
Clear error message or doc to remind user 'pip install .'

**Comment by binmakeswell** [comment]
TODO ckpt 
Seriously affect the actual use of the user, need to be fixed as a priority

**Comment by ht-zhou** [comment]
> > Hi, I found this [picture](https://raw.githubusercontent.com/hpcaitech/public_assets/main/applications/chatgpt/experience.jpg) a little confusing. I think if there is some description, it would be much better.
> 
> Yes, at least tell the readers: the green part is the trainable parameters and the rest parts are frozen parameters.

I've already completed a detailed doc to explain about the training process, it will be released soon.

**Comment by binmakeswell** [comment]
We have updated a lot. Please check https://github.com/hpcaitech/ColossalAI/tree/main/applications/Chat
This issue was closed due to inactivity. Thanks.
